### PR TITLE
IGNITE-20371 Move handlers of index-related commands from CatalogManager

### PR DESCRIPTION
**Pull request by korlov42**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractCreateIndexCommand.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    @Override
+    public List<UpdateEntry> get(Catalog catalog) {
+        CatalogSchemaDescriptor schema = schemaOrThrow(catalog, schemaName);
+
+        ensureNoTableOrIndexExistsWithGivenName(schema, indexName);
+
+        CatalogTableDescriptor table = tableOrThrow(schema, tableName);
+
+        assert columns != null;
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractCreateIndexCommand.java:73]: I hate when right side of code editor in IDEA looks like Christmas tree. This assertion makes IDEA a little bit happier

[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractCreateIndexCommand.java:73]: OMG, we are doing the IDEA happy. It's good enoght reason to have it )

**Review by ygerzhedovich — APPROVED** [review]

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogTableDescriptor.java [hunk 2]
```diff
@@ -96,6 +101,10 @@ public int zoneId() {
         return zoneId;
     }
 
+    public int primaryKeyId() {
+        return primaryKeyId;
```
Comment (xtern) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogTableDescriptor.java:105]: * Personally I prefer `pkIndexId` for the variable name and `primaryKeyIndexId()` for this method 
* It looks strange that we have pkColumns and pkIndexId at the same time, as far as I understand, we can always check that it is a PK index if it consists of the 'pkColumns', right? ...Or vice-versa, we should be able to get a list of pk columns by pkIndexId.

**Review by xtern — APPROVED** [review]

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogTableDescriptor.java (hunk 2 above)
Comment (korlov42) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogTableDescriptor.java]: renamed

**Pull request merged**